src/transformers/integrations/peft.py

```diff
@@ -333,7 +356,34 @@ def active_adapter(self) -> str:
 
         for _, module in self.named_modules():
             if isinstance(module, BaseTunerLayer):
-                return module.active_adapter
+                active_adapters = module.active_adapter
+                break
+
+        # For previous PEFT versions
+        if isinstance(active_adapters, str):
+            active_adapters = [active_adapters]
+
+        return active_adapters
+
+    def active_adapter(self) -> str:
```

Comment:
As discussed let's just deprecate `active_adapter(...)` in favor of `active_adapters`

```suggestion
    def active_adapter(self) -> str:
        # deprecate ...
        return self.active_adapters[0]
```